packages/fast-usdc-contract/src/exos/status-manager.ts

```diff
@@ -82,16 +59,37 @@ export const prepareStatusManager = (
   { marshaller, routeHealth }: StatusManagerPowers,
 ) => {
   /**
-   * Keyed by a tuple of the Noble Forwarding Account and amount.
+   * Keyed by Noble Forwarding Account
+   *
+   * Transactions that are `Advanced`, `AdvanceFailed`, or `AdvanceSkipped`
+   * and are awaiting a mint.
    */
-  const pendingSettleTxs: MapStore<PendingTxKey, PendingTx[]> = zone.mapStore(
+  const pendingSettleTxs: MapStore<NobleAddress, PendingTx[]> = zone.mapStore(
     'PendingSettleTxs',
     {
       keyShape: M.string(),
       valueShape: M.arrayOf(PendingTxShape),
     },
   );
 
+  const { addPendingSettleTx } = makeSettlementMatching(pendingSettleTxs);
+
+  /**
+   * Migrate `pendingSettleTxs` to be keyed by NFA instead of the tuple of NFA
+   * and amount.
+   *
+   * Uses `zone.makeOnce` as a cheap abstraction for "only do this once".
+   */
+  zone.makeOnce('migratePendingSettleStore', () => {
+    const current = [...pendingSettleTxs.values()]; // XXX copy necessary?
```

Comment:
Oh I see, the iterator might have nothing after `.clear()` is called. I think that's correct. Please do verify and then document decisively